model/project.go

```diff
@@ -372,6 +375,44 @@ type ProjectTask struct {
 	PatchOnly *bool `yaml:"patch_only,omitempty" bson:"patch_only,omitempty"`
 }
 
+type LoggerConfig struct {
+	Type         string `yaml:"type" bson:"type"`
+	SplunkServer string `yaml:"splunk_server,omitempty" bson:"splunk_server,omitempty"`
+	SplunkToken  string `yaml:"splunk_token,omitempty" bson:"splunk_token,omitempty"`
+}
+
+func (c *LoggerConfig) IsValid() error {
+	catcher := grip.NewBasicCatcher()
+	if !util.StringSliceContains(ValidLogSenders, c.Type) {
+		catcher.Add(errors.Errorf("%s is not a valid log sender", c.Type))
+	}
+	if c.Type == string(SplunkLogSender) && c.SplunkServer == "" {
+		catcher.Add(errors.New("Splunk logger requires a server URL"))
+	}
+	if c.Type == string(SplunkLogSender) && c.SplunkToken == "" {
+		catcher.Add(errors.New("Splunk logger requires a token"))
+	}
+
+	return catcher.Resolve()
+}
+
+type LogSender string
+
+const (
+	EvergreenLogSender LogSender = "evergreen"
```

Comment:
Will passing a `LogSender` instead of a `string` matter elsewhere? If not, you could simplify some of this code by just using string constants.

Reply:
It's a bit more type safety. I prefer it